From ebiederm@xmission.com  Fri Nov  6 14:49:56 2009
From: ebiederm@xmission.com (Eric W. Biederman)
Date: Wed, 04 Nov 2009 02:50:28 -0800
Subject: Driver core: Don't remove kobjects in device_shutdown.
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: <kay.sievers@vrfy.org>
Message-ID: <m1ws2636ob.fsf@fess.ebiederm.org>


device_shutdown is defined to just shutdown the hardware and to not
clean up any kernel data structures.  Therefore don't put the kobjects
for /sys/dev and /sys/dev/block and /sys/dev/char.

This ensures we don't remove /sys/dev/block and /sys/dev/char while
we still have symlinks from there to the actual devices.

Acked-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Eric W. Biederman <ebiederm@aristanetworks.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/core.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -1730,8 +1730,5 @@ void device_shutdown(void)
 			dev->driver->shutdown(dev);
 		}
 	}
-	kobject_put(sysfs_dev_char_kobj);
-	kobject_put(sysfs_dev_block_kobj);
-	kobject_put(dev_kobj);
 	async_synchronize_full();
 }
